### PR TITLE
Avoid crash if peer goes missing

### LINKED ISSUE #4013: Beacon node error: Invalid memory address or nil pointer dereference
I got this error when trying to sync the beacon node on windows. Please see the attached screenshot. After i got the error i started the client over again, and the sync continued where it left of. I'm now at block 17000 without getting the same error any more times, it only happened once.

unknown

### PR DESCRIPTION
**Pull request by mcdee**
round-robin ranges over a list of peers.  If a peer goes missing between fetching the list and fetching its status `peerstatus.Get()` will return `nil`.  This adds a simple check to pass over such peers and avoid crashing.

Resolves #4013

### CODE REVIEW

**Comment by mcdee** [comment]
Note this does not yet fully resolve the issue; working on it.

**Comment by prestonvanloon** [comment]
Do we need to block? this is still an improvement, potentially

**Comment by nisdas** [comment]
I think @mcdee still wanted to add in more fixes to this.

**Comment by mcdee** [comment]
The current patch just pushes the panic a little further down the processing.  Thee are another couple of changes I want to test to see if they manage to fix the issue properly.

**Comment by mcdee** [comment]
Okay, so one of the additional changes just catches another area where the peer can vanish between us fetching the list and querying its status.

The second part is more open to discussion.  At current it checks to see if there are any peers, and if not it waits for a few seconds and loops.  This *should* catch the situation we have seen where all peers vanish (e.g. a network hiccup) whilst we are processing blocks, however although I've been running 2-3 nodes on different hardware for a couple of days now this hasn't occurred so would appreciate others taking a look and consideration on this piece.

File: beacon-chain/sync/initial-sync/round_robin.go [hunk 1]
```diff
@@ -48,6 +49,11 @@ func (s *InitialSync) roundRobinSync(genesis time.Time) error {
 	// Step 1 - Sync to end of finalized epoch.
 	for s.chain.HeadSlot() < helpers.StartSlot(highestFinalizedEpoch()+1) {
 		root, finalizedEpoch, peers := bestFinalized()
+		if len(peers) == 0 {
+			log.Warn("No peers; waiting for reconnect")
+			time.Sleep(noPeersWaitTime)
```
Comment (nisdas) [inline beacon-chain/sync/initial-sync/round_robin.go]: can we just use the same refresh period as `refereshTime` ? its 6 seconds, so would serve the same purporse

Comment (mcdee) [reply beacon-chain/sync/initial-sync/round_robin.go]: Yeah the patches crossed so it wasn't there when I picked my name.  Will use that instead.

**Pull request merged**